backend/api/api.go

```diff
@@ -13,6 +14,15 @@ import (
 	log "github.com/sirupsen/logrus"
 )
 
+func isUserLoggedIn(c *fiber.Ctx, store *session.Store) (bool, error) {
+	sess, err := store.Get(c)
+	if err != nil {
+		return false, err
+	}
+	key := sess.Get("api_key")
+	return key != nil, nil
+}
+
```

Comment:
This is essentially the same as the `getSessionApiKey()` function, so one could use that function instead.  It is what the other endpoints are already doing.

Reply:
After discussion, we agreed that we might keep the `isUserLoggedIn()` as a boolean function but rewrite it in terms of the `getSessionApiKey()` function (i.e., it would call that function and test for error). This avoids code duplication and allows us to test whether the user is logged in in situations where we may not need the API key.